MoltenVK/MoltenVK/Commands/MVKCommandEncoderState.mm

```diff
@@ -464,6 +466,11 @@ static void assertMissingSwizzles(bool needsSwizzle, const char* stageName, MVKV
 
 void MVKGraphicsResourcesCommandEncoderState::encodeImpl() {
 
+    bool fullImageViewSwizzle = false;
+    MVKPipeline* pipeline = _cmdEncoder->_graphicsPipelineState.getPipeline();
+    if (pipeline)
+        fullImageViewSwizzle = pipeline->fullImageViewSwizzle();
```

Comment:
Following on my question in `MVKPipeline.h` below...is the purpose of this call and `fullImageViewSwizzle` local var to measure the state of `_device->_pMVKConfig->fullImageViewSwizzle` when the pipeline was created...and not when the draw/dispatch command is invoked? Is this because the status of that config value changes dynamically on some games you are testing?

Reply:
No.

What I think is happening is:

1) A game creates some image views with non-default swizzles. These don't warn, because `fullImageViewSwizzle` is on.
1) The game creates a pipeline with `fullImageViewSwizzle` on, with a shader which uses storage image views, but no sampled image views.
1) The game sets these image views into a descriptor set, then binds the descriptor set with the pipeline.
1) When it's time to draw/dispatch, MoltenVK sees that some of the image views have non-default swizzles, but the pipeline didn't request an aux buffer, because the shader doesn't use sampled images. So it warns about *all* the image views, even the ones with default swizzles, even though the shader never actually samples from any of the image views.

Reply:
Ah! Okay. Thanks for the explanation.

The Vulkan spec includes:

> ...VkComponentMapping...This remapping must be identity for storage image descriptors,...

Do you interpret that to mean that only sampled images are to be swizzled? Or would we also need to apply swizzles in SPIRV-Cross to texture reads, etc?

If only sampled image views are to be swizzled...isn't the app barred from setting a non-identity swizzle on an image view that will be used without a sampler?

Reply:
I've heard back. The log messages were all eliminated. (Unfortunately, it didn't help with the fence timeout...) I can see that the converted shaders that do sampling have swizzle constants in them, so I know that `fullImageViewSwizzle` is definitely on.

I'm pretty sure that DXVK (for one) is being lazy. I know that it creates image views right off the bat for images it creates. It also has a tendency to create dummy images for unbound resources, and give them (0, 0, 0, 0) swizzles.

Reply:
Thanks for the update.